src/org/zaproxy/zap/extension/websocket/manualsend/WebSocketPanelSender.java

```diff
@@ -80,14 +86,44 @@ public void cleanup() {
     }
     
     private WebSocketProxy getDelegate(Integer channelId) {
-    	if (!connectedProxies.containsKey(channelId)) {
+    	if (!connectedProxies.containsKey(channelId) && !closedProxies.containsKey(channelId)) {
     		logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
     		throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel") 
     				+ " " + Constant.messages.getString("websocket.manual_send.fail"));
-    	}
+    	}else if (closedProxies.containsKey(channelId)){
+			logger.warn("Selected WebSocket channel is not connected. Unable to send manual crafted message!");
+			throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.disconnected_channel")
+					+ " " + Constant.messages.getString("websocket.manual_send.use_reopen"));
+		}
         return connectedProxies.get(channelId);
     }
 
+	/**
+	 * Reopen a new WebSocket Connection same as the channel it gets. Make use of {@link WebSocketReopenSend} to establish the new connection.
+	 * @param channelId The connection to re-open
+	 * @param newSecWebSocketKey New Sec-WebSocket-Key, null to use the same key as the initial connection
+	 * @param useTrackingSessionState true to follow redirections
+	 * @param followRedirects useTrackingSessionState false for stateless http request
+	 * @return Returns false if fails to re-open the connection
+	 */
+    public boolean reOpenChannel(Integer channelId, String newSecWebSocketKey, boolean useTrackingSessionState, boolean followRedirects) {
+        return webSocketReopenSend.reEstablishConnection(getWebSocketProxyFromChannelID(channelId), newSecWebSocketKey, useTrackingSessionState, followRedirects);
+	}
+
+	public String getSecWebKeyFrom(Integer channelId) throws DatabaseException, HttpMalformedHeaderException {
+        return getWebSocketProxyFromChannelID(channelId).getHandshakeReference().getHttpMessage().getRequestHeader().getHeader("Sec-WebSocket-Key");
+	}
+
+	public WebSocketProxy getWebSocketProxyFromChannelID(Integer channelId){
+        if(closedProxies.containsKey(channelId)){
+            return closedProxies.get(channelId);
+        }else if(connectedProxies.containsKey(channelId)){
+            return connectedProxies.get(channelId);
+        }else{
+            throw new IllegalArgumentException(Constant.messages.getString("websocket.manual_send.fail.invalid_channel"));
+        }
+    }

```

Comment:
Something looks off with the indentation between L107 and L126, I'm guessing it's because of mixed chars (spaces/tabs).